### PR TITLE
Add account service gauges for connection count and time since last sync

### PR DESCRIPTION
**Pull request by lightningrob**

### CODE REVIEW

File: ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java [hunk 1]
```diff
@@ -110,4 +113,24 @@ void trackAccountDataInconsistency(CompositeAccountService compositeAccountServi
...
+   * @param gauge the function returning the elapsed time.
+   */
+  void trackTimeSinceLastSync(Gauge<Integer> gauge) {
+    timeInSecondsSinceLastSyncGauge = gauge;
+    metricRegistry.register(MetricRegistry.name(MySqlAccountService.class, "TimeInSecondsSinceLastSync"),
+        timeInSecondsSinceLastSyncGauge);
+  }
+
+  /**
+   * Tracks the number of seconds elapsed since the last database sync.
```
Comment (Arun-LinkedIn) [inline ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java:128]: nit: Typo with description.

**Review by Arun-LinkedIn — APPROVED** [review]

File: ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java [hunk 2]
```diff
@@ -81,6 +83,8 @@ public MySqlAccountService(AccountServiceMetrics accountServiceMetrics, MySqlAcc
         throw e;
       }
     }
+    accountServiceMetrics.trackTimeSinceLastSync(this::getTimeInSecondsSinceLastSync);
+    accountServiceMetrics.trackContainerCount(this::getContainerCount);
```
Comment (jsjtzyy) [inline ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java:87]: minor: how about moving these two to end  of constructor after cache is initialized and initial fetch is performed?

File: ambry-api/src/main/java/com/github/ambry/account/Account.java [hunk 3]
```diff
@@ -284,6 +284,13 @@ public Container getContainerByName(String containerName) {
     return Collections.unmodifiableCollection(containerIdToContainerMap.values());
   }
 
+  /**
+   * @return the number of containers in this account.
+   */
+  public int getContainerCount() {
+    return containerIdToContainerMap.size();
+  }
+
```
Comment (jsjtzyy) [inline ambry-api/src/main/java/com/github/ambry/account/Account.java:293]: Not related to this change but it seems `HELIX_ACCOUNT_SERVICE_ACCOUNT_NAME` in this file can be removed.

**Review by jsjtzyy — APPROVED** [review]
LGTM

File: ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java (hunk 1 above)
Comment (lightningrob) [reply ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java:128]: sorry, what's the typo?

Comment (jsjtzyy) [reply ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java:128]: The description is copied from preivous method, should be updated.

File: ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java (hunk 2 above)
Comment (lightningrob) [reply ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java:87]: I put them there so the gauge values (especially timeSinceLastSync would get populated during the initial fetch.

File: ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java (hunk 1 above)
Comment (lightningrob) [reply ambry-account/src/main/java/com/github/ambry/account/AccountServiceMetrics.java:128]: oh right, will fix.

File: ambry-api/src/main/java/com/github/ambry/account/Account.java (hunk 3 above)
Comment (lightningrob) [reply ambry-api/src/main/java/com/github/ambry/account/Account.java:293]: It could be used by LI code, but I guess we'll find out when we integrate.

**Review by jsjtzyy — APPROVED** [review]

**Pull request merged**